client/trino-cli/src/main/java/io/trino/cli/QueryRunner.java

```diff
@@ -93,6 +101,7 @@ public QueryRunner(
         setupHttpProxy(builder, httpProxy);
         setupBasicAuth(builder, session, user, password);
         setupTokenAuth(builder, session, accessToken);
+        setupExternalAuth(builder, session, externalAuthentication, sslSetup);
```

Comment:
Do we need any kind of validation? Like external auth does not go with basic auth or krb?

Reply:
We don't do this for any kind of other auth and the assumption is that configuring multiple authentication methods is not supported.